### ISSUE TITLE
[CI] Changes to workflows are not tested on PRs

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector#9676]
During the v0.96.0 release cycle I had to file #9675 to fix the release pipeline. This is because #9409 accidentally broke the workflow. Mistakes happen, and it's important that we test workflows so that we catch these issues during the PR process.

Github seems to have something builtin, so probably we can reuse an action of theirs for this.

**Comment by Sanket-0510** [comment]
Hey I know I missed a lot of things while writing prepare-release workflow file. But even I could not notice that I have missed onto a single `s` in the name. This is because nothing was there to test the syntax for the workflow file, I should have looked at the file carefully before pushing the commit.

**Comment by Sanket-0510** [comment]
> @Sanket-0510 Not your fault at all! To be crystal clear: this is a problem with our CI and how we do testing and not with any individual contributor. Mistakes happen (I myself am very prone to making a lot of mistakes) and the focus should not be on "being careful" but on having CI and tooling that allows us to catch these mistakes.

@mx-psi will need to write on more CI to test the linting of all the workflow files right?

**Comment by mx-psi** [comment]
@Sanket-0510 Yes, in order to solve this issue this is one of the things that would help

**Comment by KenEkanem** [comment]
Hello @mx-psi and @Sanket-0510. 
I am an outreachy intern,  I would like to contribute to resolving this issue by adding a CI job that performs linting on all workflow files. can you assign me to this?

**Comment by mx-psi** [comment]
@KenEkanem Assigned to you, a linter sounds like a good approach for this :)

**Comment by KenEkanem** [comment]
Hey @mx-psi before I open the PR I wanted to know should I allow the job to continue even if there are errors in the workflow files or show the job run as failed if any errors are detected.

**Comment by mx-psi** [comment]
CI should fail if an error is detected

**Comment by TylerHelmuth** [comment]
We ran into broken workflows again during the v0.123.0 release. During this release the automated prep PRs in Core, Contrib, and Releases repos failed as did the `release` workflows for the distributions in the Releases repo.

The automated prep PRs failed due to an incorrect `v` and the release workflow for distributions failed because a `dockerhub` section was added to the goreleaser manifest files but the proper `username` was not set.

For these issues I believe the problems could have been caught prior to merging the changes by requiring the PR submitter to test the workflow change in a fork of the repo.

**Comment by priyendang** [comment]
Hi @mx-psi , I noticed this issue was previously being worked on but there hasn't been activity for a while. If it's still open, I'd be happy to take this up and work on a fix. Please let me know if that works.

**Comment by mx-psi** [comment]
Feel free to take it!

**Comment by BryanAndy** [comment]
Hi @mx-psi, I noticed this issue was previously being assigned, but there hasn't been any activity for a while. Is there a PR open fr it? If not, I would like to work on a solution for it. If that works, let me know.

**Comment by om7057** [comment]
Hi @mx-psi, I see actionlint was added via #12721 which covers syntax linting. Given @TylerHelmuth's comment about logic errors still causing release failures, is there a specific remaining gap you'd want addressed here? Happy to look into it if the scope is clear.
